### PR TITLE
Map taxes to channel

### PR DESCRIPTION
**Pull request by riteshshrv**

### CODE REVIEW

File: channel.py [hunk 1]
```diff
@@ -98,6 +98,7 @@ class SaleChannel(ModelSQL, ModelView):
         fields.Many2One('party.party', 'Company Party'),
         'on_change_with_company_party'
     )
+    taxes = fields.One2Many("sale.channel.tax", "channel", "Tax")
```
Comment (tarunbhardwaj) [inline channel.py]: String Taxes

File: channel.py [hunk 2]
```diff
@@ -688,6 +691,24 @@ def update_order_status(self):
...
         )
 
+    def get_tax(self, name, rate):
+        """
+        Search for an existing Tax record by matching name and rate.
+        If found return its active record else raise user error.
+        """
+        TaxMapping = Pool().get('sale.channel.tax')
+
+        mapped_tax = TaxMapping.search([
```
Comment (tarunbhardwaj) [inline channel.py]: use `mapped_tax, = ...`

File: channel.py [hunk 3]
```diff
@@ -688,6 +691,24 @@ def update_order_status(self):
...
+            ('name', '=', name),
+            ('rate', '=', rate),
```
Comment (tarunbhardwaj) [inline channel.py]: channel = self

File: channel.py [hunk 4]
```diff
@@ -688,6 +691,24 @@ def update_order_status(self):
...
+        ])
+        if mapped_tax:
+            return mapped_tax[0]
```
Comment (tarunbhardwaj) [inline channel.py]: mapped_tax.tax

**Comment by tarunbhardwaj** [comment]
Add tests

File: channel.py [hunk 5]
```diff
@@ -816,3 +839,26 @@ class ChannelOrderState(ModelSQL, ModelView):
...
+        cls._error_messages.update({
+            'unique_tax_rate_per_channel':
+            "Tax rate and name must be unique per channel"
+        })
+        cls._sql_constraints += [
+            ('unique_tax_percent', 'UNIQUE(channel, name, rate)',
+             'unique_tax_rate_per_channel')
+        ]
+
+    name = fields.Char("Name", required=True)
```
Comment (prakashpp) [inline channel.py]: field definition always on top

**Comment by tarunbhardwaj** [comment]
Merged It is some issue with github PR is still open.

**Pull request closed without merging**